src/NServiceBus.Testing.Tests/Handler/HandlerTests.cs

```diff
@@ -61,6 +61,28 @@ public void SendMessageWithMultiIncomingHeaders()
         }
 
         [Test]
+        public void ShouldBeAbleToConfigureMessageHandlerContext()
+        {
+            var messageId = Guid.NewGuid().ToString();
+            var replyToAddress = "0118 999 881 999 119 725 3";
+            var handler = new ContextAccessingHandler();
+            TestableMessageHandlerContext contextInstance = null;
+
+            Test.Handler(handler)
+                .ConfigureHandlerContext(c =>
```

Comment:
I think the API alignment is the key here. Otherwise I would have voted for @mat-mcloughlin proposal.